### ISSUE TITLE
overflow-menu : Oververflow-menu-body position is wrong 

### ISSUE THREAD

**Issue by rccc** [carbon-design-system/carbon-for-ibm-dotcom#9827]
### Description

*overflow-menu-body* should be inside *oveflow-menu*, but once *oveflow-menu* is open, i noticed *overflow-menu-body is outside* and then position on screen is wrong. 

### Component(s) impacted

overflow-menu

![Capture d’écran du 2022-12-15 14-54-27](https://user-images.githubusercontent.com/122856/207880602-d9aa6c75-86fb-4df5-9a24-03fd77ab96ec.png)

### Browser

Chrome, Firefox

### Carbon for IBM.com version

1.21.0

### Severity

Severity 1 = The design is broken in a critical way that blocks users from completing tasks or damages the brand. Affects major functionality, no workaround.

### Application/website

https://codesandbox.io/s/github/carbon-design-system/carbon-web-components/tree/main/examples/codesandbox/basic/components/pagination

### Package

@carbon/ibmdotcom-web-components

### CodeSandbox example

https://codesandbox.io/s/github/carbon-design-system/carbon-web-components/tree/main/examples/codesandbox/basic/components/pagination

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues

**Comment by techtolentino** [comment]
Same thing happens for us, when trying to use it... 

<img width="1728" alt="Screenshot 2023-06-02 at 1 32 51 PM" src="https://github.com/carbon-design-system/carbon-for-ibm-dotcom/assets/6276074/a006ddbf-2153-4111-86f8-8d067e34e28d">

**Comment by kennylam** [comment]
@rccc @techtolentino Can you share repros?

**Comment by rccc** [comment]
@kennylam 

I am going to check what i can share. Let me a few days.

Eric

**Comment by techtolentino** [comment]
@rccc - thanks for responding. Don't worry about it. I figured out ☝🏼 this issue, but then ran into other issues within our own application that makes this particular component unusable (for our use case). 

Thanks!

**Comment by dave-wood** [comment]
Hi - as discussed with @ariellalgilmore I'm seeing a weird bug when using an overflow menu within a data-table to do filtering on the table. See this example:
https://codesandbox.io/p/devbox/stoic-wescoff-k4wlrm?file=%2Fsrc%2Findex.js%3A18%2C1

If I reload the preview panel, scroll down with the scroll bar until Example table is in the middle of the screen, then click the filter icon (next to the Search icon) in the Example Table, it opens an overflow menu, but jumps the page to the top.

Interestingly, I've discovered whilst using codesandbox, that it only started jumping to the top once I added the pagination for the table at the bottom! Specifically, when I added the select/index.js component to make the page number drop down work. So even though the affected table doesn't have pagination, the presence of pagination on a different table causes the problem.

**Comment by aaronmg241** [comment]
Any fixes for this? The component has the described behaviour no matter its location (within a table, direct descendant of body tag, etc). I was able to reproduce in:
https://codesandbox.io/p/sandbox/naughty-curran-rx95n3?file=%2Findex.html as well. 

The only fix I have right now is to use a workaround where we manually set the top attribute using a MutationObserver whenever the top attribute is attempted to be changed. This is obviously unsustainable, but the component is otherwise unusable.
